collect_app/src/main/AndroidManifest.xml

```diff
@@ -118,7 +118,9 @@ the specific language governing permissions and limitations under the License.
             android:name=".activities.FillBlankFormActivity"
             android:launchMode="singleTop" />
         <activity android:name=".activities.FormDownloadListActivity" />
-        <activity android:name="org.odk.collect.android.errors.ErrorActivity" />
+        <activity
```

Comment:
Do we need this here? It's declared in errors module.

Reply:
Yes so we can override the theme.

Reply:
I see so wouldn't it be better to make it possible to access the theme in other modules so that we can set it in `errors` module only?

Reply:
Personally I quite like being able to avoid using a shared component/interface here and just setting the theme in the manifest. It keeps the library modules simpler. An alternative would be doing something like `LocalizedApplication` for themes but, again, I think I prefer what we have?

Reply:
again no strong feelings so I could accept but now I'm thinking... If that ErrorActivity no longer extends CollectAbstractActivity (responsible for updating themes), what happens if we have `Use device theme` selected in settings and for example we have that activity opened in light mode and then we set to dark mode and reopen the app? Will the theme be updated properly?

Reply:
Great question. We do call `setTheme` in `CollectAbstractActivity` still, but the actual switching between light, dark and system happens in `MainMenuActivity#onCreate` (when returning from settings, switching projects or opening the app). 

The call in `CollectAbstractActivity` could actually be replaced with adding `theme` attributes to all our Activity declarations in the manifest (or potentially the [`application` `theme` attribute](https://developer.android.com/guide/topics/manifest/application-element#theme) which we're not using for some reason).

Reply:
Ok so as I understand it's acceptable that in order to update the theme we need to go though `MainMenuActivity`? I'm asking because since we had this in `CollectAbstractActivity` which we used as a base activity for other activities I think we wanted to  have that theme updating everywhere.

If it's the expected behavior I think we should have a pr that cleans that up (removes that code from `CollectAbstractActivity`) because it's confusing - of course it might be a separate pull request.

Reply:
>  I'm asking because since we had this in CollectAbstractActivity which we used as a base activity for other activities I think we wanted to have that theme updating everywhere.

We're no longer updating the actual Android theme being used any longer - we just change the night mode setting that appcompat uses (using `AppCompatDelegate.setDefaultNightMode`) and we only need to do this when UI "theme" setting has changed. The night mode then dictates which resources get loaded (`-night` or not).

> If it's the expected behavior I think we should have a pr that cleans that up (removes that code from CollectAbstractActivity) because it's confusing - of course it might be a separate pull request.

I think you're right - it's confusing in the current state. I can do that clean up here as it's a refactor PR anyway and it'd be nice to not have to file an issue etc!

Reply:
Done!